autorest/azure/auth/authfile.go

```diff
@@ -0,0 +1,124 @@
+package auth
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"unicode/utf16"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/dimchansky/utfbom"
+)
+
+// Auth includes authentication details for ARM clients
+type Auth struct {
```

Comment:
This type name stutters.  I don't have enough context here to suggest a better name.

Reply:
Fixed